src/radio.c

```diff
@@ -0,0 +1,250 @@
+/**
+ * collectd - src/radio.c
+ * Copyright (C) 2016 rinigus
+ *
+ *
+ The MIT License (MIT)
+
+ Permission is hereby granted, free of charge, to any person obtaining
+ a copy of this software and associated documentation files (the
+ "Software"), to deal in the Software without restriction, including
+ without limitation the rights to use, copy, modify, merge, publish,
+ distribute, sublicense, and/or sell copies of the Software, and to
+ permit persons to whom the Software is furnished to do so, subject to
+ the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ * Authors:
+ *   rinigus <http://github.com/rinigus>
+
+ Radio switches are monitored, as exposed by rfkill interface in Linux
+ 
+ Sysfs description: https://www.kernel.org/doc/Documentation/ABI/stable/sysfs-class-rfkill
+ 
+ 
+**/
+
+#include "collectd.h"
+#include "common.h"
+#include "plugin.h"
+
+#define SYSFS_ROOT "/sys/class/rfkill/"
+#define BFSZ 512
+
+static int num_rfkill = 0;
+
+static int radio_init (void)
+{
+	int status;
+	char filename[256];
+
+	num_rfkill = 0;
+  
+	while (1)
+	{
+		status = ssnprintf (filename, sizeof (filename),
+							SYSFS_ROOT "rfkill%d/"
+							"hard", num_rfkill);
+      
+		if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+			break;
+      
+		if (access (filename, R_OK))
+			break;
+      
+		num_rfkill++;
+	}
+  
+	INFO ("radio plugin: Found %d radio%s", num_rfkill,
+		  (num_rfkill == 1) ? "" : "s");
+  
+	if (num_rfkill == 0)
+		plugin_unregister_read ("radio");
+  
+	return (0);
+}
+
+
+static void radio_submit (const char *type, const char *plugin_instance, const char *type_instance, gauge_t value)
+{
+	value_t values[1];
+	value_list_t vl = VALUE_LIST_INIT;
+  
+	values[0].gauge = value;
+  
+	vl.values = values;
+	vl.values_len = 1;
+	sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+	sstrncpy (vl.plugin, "radio", sizeof (vl.plugin));
+	sstrncpy (vl.plugin_instance, plugin_instance, sizeof (vl.plugin_instance));
+	sstrncpy (vl.type, type, sizeof (vl.type));
+	sstrncpy (vl.type_instance, type_instance, sizeof (vl.type_instance));
+  
+	plugin_dispatch_values (&vl);
+}
+
+
+static _Bool get_value(const char *fname, int *value)
+{
+	FILE *fh;
+	char buffer[BFSZ];
+
+	if ((fh = fopen(fname, "r")) == NULL)
+		return (0);
+
+	if ( fgets(buffer, sizeof(buffer), fh) == NULL )
+    {
```

Comment:
indentation (see above)

no space after ( with the exception of 

if ( !some_condition )

consistently through the text to improve visibility of !